### DISCUSSION TITLE
My custom BlueBuild image has been failing to build for the past week or so

### DISCUSSION THREAD

**Discussion by alexispurslane** [blue-build/.github#38] [Support] [answered]
They all seem to be random package dependency version conflict issues, not anything inherent to what I've done, but I'm concerned about not getting updates for so long. Is there something I can do to fix this?

https://github.com/alexispurslane/silverblue-alexis/actions/workflows/build.yml

**Comment by fiftydinar** [comment, marked as answer]
Schily package seems to conflict with coreutils for some reason.
Other than removing it, I don't see the solution to build failures.

**Reply by alexispurslane** [reply]
I was able to resolve the problem by removing GNOME Bottles, which was apparently what was pulling Schily in. Unfortunately, now that Fedora 41 is out, I'm running into [***yet another***](https://github.com/alexispurslane/silverblue-alexis/actions/runs/11645126384/job/32427810024) problem. I'm not even doing anything strange or convoluted with my image, so Im beginning to have second thoughts about how stable BlueBuild is. Is it just not for me?

**Reply by xynydev** [reply]
BlueBuild itself is pretty stable, your issues are not caused by it. `rpm-ostree` is not stable. Fedora does not consider the features used by BlueBuild, Universal Blue, etc. stable, until we can use `dnf` and `bootc` to manage our systems.

> Is it just not for me?

Depends. If you're running into issues adding packages with BlueBuild, you would run into the same issues without. Is atomic Fedora for you? (or rather, is it stable enough for you?) -- Those questions you kind of have to examine for yourself.

I personally do not run in to package conflicts and other issues, because I consciously choose not to use packages that cause those issues. This is not an option for everyone unfortunately.

But for many programs, a Flatpak or a container / distrobox is simply the better option. Adding everything you need as system RPM packages is not recommended. That's true _even when using custom images_, not only when just running an unmodified Fedora image.

It's true that the whole field is currently moving fast, and especially the Universal Blue org is exploring `systemd-sysext` as a package distribution mechanism, constantly monitoring `bootc` progress, etc. There's innovation still yet to be done, features yet to be finished, and annoying issues like these yet to be solved.

But if you do decide that this project / paradigm is not for you right now, I'm pretty positive that in a few years the issues described here will be history.

**Reply by xynydev** [reply]
Plus if you're curious to try out if VanillaOS packages and package management fits your needs more, I have a testing repo for BlueBuild x VanillaOS (that nobody else has tested yet, so the interest and priority for this seems pretty low right now...)

https://github.com/xynydev/vanillin

**Comment by alexispurslane** [comment]
Alright, if there's nothing else I can do, I'll look into removing whatever is pulling the schily package in.
— alexis (she/her)
Sent from Proton Mail Android
-------- Original Message --------On 10/16/24 7:49 AM, fiftydinar  wrote:
Schily package seems to conflict with coreutils for some reason.
Other than removing it, I don't see the solution to build failures.

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you authored the thread.Message ID: ***@***.***>
[
{
***@***.***": "http://schema.org",
***@***.***": "EmailMessage",
"potentialAction": {
***@***.***": "ViewAction",
"target": "https://github.com/orgs/blue-build/discussions/38#discussioncomment-10959005",
"url": "https://github.com/orgs/blue-build/discussions/38#discussioncomment-10959005",
"name": "View Discussion"
},
"description": "View this Discussion on GitHub",
"publisher": {
***@***.***": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Comment by alexispurslane** [comment]
I wrote the message above late at night after a long shift at work, in a lot of pain due to my chronic illness, and thus in a bad mood and frustrated. Thanks for responding kindly.
Upon further reflection, it's clear to me that the problems I'm having would indeed present themselves with or without BlueBuild, since they certainly do seem to be merely packaging issues. In fact, the new issue I mentioned is simply a result of the package I was trying to install not being available for Fedora 41 yet, and the other one is just packaged poorly. In which case, BlueBuild actually just *saved* me from errors that likely would have presented themselves on regular fedora and fedora atomic as well: it prevented me from having to worry about having to deal with those issues on my local machine, or get an update before it was ready based on the packages I need.
Moreover I'm very certain that the Fedora Atomic / BlueBuild model is the future of desktop Linux, and despite my overly hyperbolic reaction the other night, I've found it far more stable already than any other distro I've used, without sacrificing up to date packages.
Also, yes, I know to not layer everything — I have a very minimal set of packages layered, only things that I absolutely need to be part of my system because they simply wouldn't work as flatpaks (firmware-manager, chkrootkit, formerly bottles since I need it to have USB access), or I want them to be available on my host system (my shell and terminal emulator is emacs, so I need it, and I need fish as a completion server), as well as inside any distroboxes I might use.
— alexis (she/her)
Sent from Proton Mail Android
-------- Original Message --------On 11/5/24 8:22 AM, xyny  wrote:
BlueBuild itself is pretty stable, your issues are not caused by it. rpm-ostree is not stable. Fedora does not consider the features used by BlueBuild, Universal Blue, etc. stable, until we can use dnf and bootc to manage our systems.

Is it just not for me?

Depends. If you're running into issues adding packages with BlueBuild, you would run into the same issues without. Is atomic Fedora for you? (or rather, is it stable enough for you?) -- Those questions you kind of have to examine for yourself.
I personally do not run in to package conflicts and other issues, because I consciously choose not to use packages that cause those issues. This is not an option for everyone unfortunately.
But for many programs, a Flatpak or a container / distrobox is simply the better option. Adding everything you need as system RPM packages is not recommended. That's true even when using custom images, not only when just running an unmodified Fedora image.
It's true that the whole field is currently moving fast, and especially the Universal Blue org is exploring systemd-sysext as a package distribution mechanism, constantly monitoring bootc progress, etc. There's innovation still yet to be done, features yet to be finished, and annoying issues like these yet to be solved.
But if you do decide that this project / paradigm is not for you right now, I'm pretty positive that in a few years the issues described here will be history.

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you authored the thread.Message ID: ***@***.***>
[
{
***@***.***": "http://schema.org",
***@***.***": "EmailMessage",
"potentialAction": {
***@***.***": "ViewAction",
"target": "https://github.com/orgs/blue-build/discussions/38#discussioncomment-11154803",
"url": "https://github.com/orgs/blue-build/discussions/38#discussioncomment-11154803",
"name": "View Discussion"
},
"description": "View this Discussion on GitHub",
"publisher": {
***@***.***": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Reply by xynydev** [reply]
No problem! I checked your repo and it indeed seems that you aren't doing anything wrong, it's just the `firmware-manager` package you're installing that does not exist in the F41 version of the COPR yet. Take care!

💙
